src/libUtils/Logger.cpp

```diff
@@ -16,325 +16,264 @@
+};
+
+class StdoutSink {
+ public:
+  void forwardLogToStdout(LogMessageMover logEntry) {
+    logMessageCommon(m_stream, logEntry.get());
+    std::cout << m_stream.str();
+    m_stream.str("");
+  }
 
-Logger::Logger(const char* prefix, bool log_to_file, const char* logpath,
-               streampos max_file_size) {
-  this->m_logToFile = log_to_file;
-  this->m_maxFileSize = max_file_size;
-  this->m_logPath = logpath;
+ private:
+  std::ostringstream m_stream;
+};
 
+template <typename LogRotateSinkT>
+void AddFileSink(LogWorker& logWorker, const std::string& filePrefix,
+                 const boost::filesystem::path& filePath,
+                 int maxFileSize /*= MAX_FILE_SIZE*/) {
+  auto logFileRoot = boost::filesystem::absolute(filePath);
+  bool useDefaultLocation = false;
   try {
-    if (!boost::filesystem::create_directory(this->m_logPath)) {
-      if ((boost::filesystem::status(this->m_logPath).permissions() &
+    if (!boost::filesystem::create_directory(logFileRoot)) {
+      if ((boost::filesystem::status(logFileRoot).permissions() &
            boost::filesystem::perms::owner_write) ==
           boost::filesystem::perms::no_perms) {
-        std::cout << this->m_logPath
+        useDefaultLocation = true;
+        std::cout << logFileRoot
                   << " already existed but no writing permission!" << endl;
-        this->m_logPath = boost::filesystem::absolute("./").string();
-        std::cout << "Use default log folder " << this->m_logPath << " instead."
-                  << endl;
       }
     }
   } catch (const boost::filesystem::filesystem_error& e) {
-    std::cout << "Cannot create log folder in " << this->m_logPath
+    std::cout << "Cannot create log folder in " << logFileRoot
               << ", error code: " << e.code() << endl;
-    this->m_logPath = boost::filesystem::absolute("./").string();
-    std::cout << "Use default log folder " << this->m_logPath << " instead."
-              << endl;
+    useDefaultLocation = true;
   }
 
-  if (log_to_file) {
-    m_fileNamePrefix = prefix ? prefix : "common";
-    m_seqNum = 0;
-    newLog();
+  if (useDefaultLocation) {
+    logFileRoot = boost::filesystem::absolute("./");
+    std::cout << "Use default log folder " << logFileRoot << " instead."
+              << endl;
   }
+
+  auto sinkHandle = logWorker.addSink(
+      std::make_unique<LogRotateSinkT>(
+          filePrefix.empty() ? "common" : filePrefix, logFileRoot.c_str()),
+      &LogRotateSinkT::receiveLogMessage);
+  sinkHandle->call(&LogRotateSinkT::setMaxLogSize, maxFileSize).wait();
+  sinkHandle
+      ->call(&LogRotateSinkT::setMaxArchiveLogCount, MAX_ARCHIVED_LOG_COUNT)
+      .wait();
 }
 
-Logger::~Logger() { m_logFile.close(); }
+}  // namespace
 
-void Logger::checkLog() {
-  std::ifstream in(m_fileName.c_str(),
-                   std::ifstream::ate | std::ifstream::binary);
+const int Logger::MAX_FILE_SIZE = 1024 * 1024 * 10;  // 10MB per log file
```

Comment:
can we make it configurable via constants.xml ?

Reply:
Made the change as per the FIXME comment.